app/src/main/java/chat/rocket/android/helper/RxHelper.java

```diff
@@ -0,0 +1,23 @@
+package chat.rocket.android.helper;
+
+import java.util.concurrent.TimeUnit;
+import rx.Observable;
+import rx.Single;
+import rx.functions.Func0;
+import rx.functions.Func1;
+
+/**
+ * Rx operator and so on.
+ */
+public class RxHelper {
+  public static Func1<Observable<? extends Throwable>, Observable<?>> exponentialBackoff(
+      int maxRetryCount, long base, TimeUnit unit) {
+    return errors -> errors
+        .zipWith(Observable.range(0, maxRetryCount), (error, retryCount) -> retryCount)
```

Comment:
I think an operator like `enumerate()` is needed here.

![slice](https://cloud.githubusercontent.com/assets/11763113/22291112/0677f410-e348-11e6-8ca8-3601f1b5dbaf.png)